src/azul/service/manifest_service.py

```diff
@@ -815,15 +823,44 @@ def manifest_config(self) -> ManifestConfig:
                               combine_script='return new ArrayList(params._agg.fields)',
                               reduce_script=reduce_script)
         es_search = es_search.extra(size=0)
-        response = es_search.execute()
-        assert len(response.hits) == 0
+        start = time.time()
+        try:
+            response = es_search.execute()
+        # If timeout is exceeded the request is partitioned to reduce load on ES
+        except ConnectionTimeout:
+            logger.warning('Elasticsearch request exceeded timeout limit, retrying …')
+            fields = self._partition_search(es_search)
+        else:
+            logger.info('Elasticsearch request completed after %.003fs',
+                        time.time() - start)
+            assert len(response.hits) == 0
+            fields = response.aggregations.fields.value
         return {
             'contents': {
                 value: value.split('.')[-1]
-                for value in sorted(response.aggregations.fields.value)
+                for value in sorted(fields)
             }
         }
 
+    def _partition_search(self, es_search: Search) -> Set[str]:
+
+        def get_fields(es_search: Search) -> List[str]:
+            response = es_search.execute()
+            assert len(response.hits) == 0, response.hits
+            return response.aggregations.fields.value
+
+        requests = []
+        for prefix in list('0123456789abcdef'):
```

Comment:
use `string.hexdigits[:16]`

Reply:
> This could be a comprehension.

It was, [originally](https://github.com/DataBiosphere/azul/pull/2473#discussion_r526456488). But it was suggested a loop might be more readable.